src/Plugin.php

```diff
@@ -642,16 +643,18 @@ public static function buildUserInfo($key_prefix = 'billing', $user_id = NULL) {
     }
 
     // If a registered user has a subscriber ID already, then alfa GP/VM will
-    // reject any kind of change to the address. The submitted address will only
-    // be contained in the order confirmation email and manually processed by
-    // the customer service team. Even if the user only specified the ID in the
-    // checkout form, it has already been validated to be correct by now.
-    if (is_checkout()) {
-      $remove_address = $user_id > 0 && (!empty($address_source['billing_subscriber_id']) || get_user_meta($user_id, 'billing_subscriber_id', TRUE));
-    }
-    else {
-      $remove_address = $user_id > 0 && get_user_meta($user_id, 'billing_subscriber_id', TRUE);
-    }
+    // reject any kind of change to the address, unless it is a "dummy address";
+    // i.e., a partial address record only consisting of salutation, firstname,
+    // lastname, used for quicker trial subscription signups, which is filled up
+    // by alfa with placeholder values for all other required address fields.
+    // If addresss values are removed, the submitted address will only be
```

Comment:
1. Typo in "addresss"

1. _"If address values are removed **below**, …"_ might be more clear